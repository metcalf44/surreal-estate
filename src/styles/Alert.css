.alert-success {
    background: rgb(210, 242, 200);
    color: rgb(1, 70, 1);
    padding: .75em 1.5em;
    margin: .25em 0 .5em 0;
    border-radius: 10px;
    text-align: center;
    border: solid rgb(1, 70, 1) 1px;
    font-weight: 800;
    font-size: 1.1rem;
}

.alert-error {
    background: rgb(244, 200, 192);
    color: rgb(120, 1, 1);
    padding: .75em 1.5em;
    margin: .25em 0 .5em 0;
    border-radius: 10px;
    text-align: center;
    border: solid rgb(159, 5, 5) 1px;
    font-weight: 800;
    font-size: 1.1rem;
}