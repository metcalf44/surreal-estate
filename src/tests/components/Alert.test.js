import React from "react";
import { render } from "@testing-library/react";
import Alert from "../../components/Alert";

test("display an error message", () => {
  const { getByText } = render(<Alert message="Error!" />);

  expect(getByText(/Error/).textContent).toBe("Error!");
});

test("display a success message", () => {
  const { getByText } = render(<Alert message="Success!!" success />);

  expect(getByText(/Success/).textContent).toBe("Success!!");
});

test("if the message is not truthy it wont display", () => {
  const { asFragment } = render(<Alert message="" />);
  const alert = asFragment();

  expect(alert).toMatchSnapshot();
});

test("will display an error message", () => {
  const { getByText, asFragment } = render(<Alert message="Error!" />);
  expect(getByText(/Error/).textContent).toBe("Error!");
  expect(asFragment()).toMatchSnapshot();
});

test("will display a success message", () => {
  const { getByText, asFragment } = render(<Alert message="Success!!" />);
  expect(getByText(/Success!!/).textContent).toBe("Success!!");
  expect(asFragment()).toMatchSnapshot();
});
