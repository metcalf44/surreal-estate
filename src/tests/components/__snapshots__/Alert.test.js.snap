// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if the message is not truthy it wont display 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-error"
  >
     
  </div>
</DocumentFragment>
`;

exports[`will display a success message 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-error"
  >
    Success!!
  </div>
</DocumentFragment>
`;

exports[`will display an error message 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-error"
  >
    Error!
  </div>
</DocumentFragment>
`;
